services/121-service/src/payments/fsp-integration/excel/excel.service.ts

```diff
@@ -243,21 +250,79 @@ export class ExcelService
     return matchColumn;
   }
 
-  public async reconciliatePayments({
+  public async processReconciliationData({
+    file,
+    payment,
+    programId,
+    fspConfigs,
+  }: {
+    file: Express.Multer.File;
+    payment: number;
+    programId: number;
+    fspConfigs: ProgramFinancialServiceProviderConfigurationEntity[];
+  }): Promise<ReconciliationDto[]> {
+    const maxRecords = 10000;
+    const validatedExcelImport = await this.fileImportService.validateCsv(
+      file,
+      maxRecords,
+    );
+
+    // First set up unfilled feedback object based on import rows ..
+    const crossFspConfigImportResults: ReconciliationDto[] = [];
+    for (const row of validatedExcelImport) {
+      const resultRow = new ReconciliationDto();
+      resultRow.feedback = new ReconciliationFeedbackDto();
+      resultRow.feedback = {
+        ...row,
+        importStatus: ImportStatus.notFound,
+        referenceId: null,
+        message: null,
+      };
+      resultRow.programFinancialServiceProviderConfigurationId = undefined;
+      resultRow.transaction = undefined;
+      crossFspConfigImportResults.push(resultRow);
+    }
+
+    // .. then loop over fspConfigs to update rows where matched
+    for await (const fspConfig of fspConfigs) {
+      const matchColumn = await this.getImportMatchColumn(fspConfig.id);
+      const importResultForFspConfig = await this.reconciliatePayments({
+        programId,
+        payment,
+        validatedExcelImport,
+        fspConfig,
+        matchColumn,
+      });
+
+      // .. then loop over each row of the original import to update if a match has been found with this fspConfig
+      crossFspConfigImportResults.forEach((row, index) => {
+        const importResultForFspConfigRow = importResultForFspConfig.find(
```

Comment:
Nice that you solved the notFound problem like this.

I wonder about the performance of this a lot of records. Maybe it's better to use a hasmap lookup here instead of an array?